lib/active_merchant/billing/gateways/element.rb

```diff
@@ -222,16 +224,44 @@ def market_code(money, options)
         options[:market_code] || 'Default'
       end
 
+      def add_lodging(xml, options)
+        if lodging = options[:lodging]
+          xml.extendedParameters do
+            xml.ExtendedParameters do
+              xml.Key 'Lodging'
+              xml.Value('xsi:type' => 'Lodging') do
+                xml.LodgingAgreementNumber lodging[:agreement_number] if lodging[:agreement_number]
+                xml.LodgingCheckInDate lodging[:check_in_date] if lodging[:check_in_date]
+                xml.LodgingCheckOutDate lodging[:check_out_date] if lodging[:check_out_date]
+                xml.LodgingRoomAmount lodging[:room_amount] if lodging[:room_amount]
+                xml.LodgingRoomTax lodging[:room_tax] if lodging[:room_tax]
+                xml.LodgingNoShowIndicator lodging[:no_show_indicator] if lodging[:no_show_indicator]
+                xml.LodgingDuration lodging[:duration] if lodging[:duration]
+                xml.LodgingCustomerName lodging[:customer_name] if lodging[:customer_name]
+                xml.LodgingClientCode lodging[:client_code] if lodging[:client_code]
+                xml.LodgingExtraChargesDetail lodging[:extra_charges_detail] if lodging[:extra_charges_detail]
+                xml.LodgingExtraChargesAmounts lodging[:extra_charges_amount] if lodging[:extra_charges_amount]
+                xml.LodgingPrestigiousPropertyCode lodging[:prestigious_property_code] if lodging[:prestigious_property_code]
+                xml.LodgingSpecialProgramCode lodging[:special_program_code] if lodging[:special_program_code]
+                xml.LodgingChargeType lodging[:charge_type] if lodging[:charge_type]
+              end
+            end
+          end
+        end
+      end
+
       def add_terminal(xml, options)
         xml.terminal do
           xml.TerminalID options[:terminal_id] || '01'
+          xml.TerminalType options[:terminal_type] if options[:terminal_type]
```

Comment:
Just wondering, since TerminalType and PartialApprovedFlag are the only net new fields here in this block, do they need to have defaults like the others? Do we know what happens if they are not provided in the request?

Reply:
...in retrospect I realize this was perhaps a silly question, since the fields have not been present in the request before now 😁 

Do we have any indication of _when_ they should be used that we could document somewhere? Do  we have the ability to test adding the partial approved flag?

Reply:
hey @jcreiff yep! So I had to add a couple of new fields because the lodging fields require them. The sandbox is pretty limited but these specifics would need to be documented.